OpenSim/Simulation/Model/Frame.cpp

```diff
@@ -87,6 +87,16 @@ void Frame::addMeshGeometry(const std::string& aGeometryFileName, const SimTK::V
     addGeometry(geom);
 }
 
+
+void Frame::attachGeometry(const OpenSim::Geometry& geom, const SimTK::Vec3 scale)
+{
+    SimTK::ClonePtr<Geometry> clone = SimTK::ClonePtr<Geometry>(geom);
+    clone->set_scale_factors(scale);
+    clone->setFrameName(getName());
+    addGeometry(clone.updRef());
```

Comment:
Oh I see. Then this looks good. The ClonePtr constructor you use will make a copy of `geom`.